tasks/create-snapshot-task.yaml

```diff
@@ -0,0 +1,156 @@
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: create-snapshot
+spec:
+  description: Creates a snapshot based on the component names, image refs and Git information passed to the task.
+  params:
+  - name: COMPONENTS
+    description: A stringified JSON array with information about components with name, container image ref, git repo and git revision for each.
+    type: string
+  - name: PRODUCT_VERSION
+    description: Product version to record as part of the Snapshot's name to be able to identify snapshots more easily. Pass here the result from determine-image-tag for the main image.
+    type: string
+  results:
+  - name: SNAPSHOT_NAME
+    description: Name of the snapshot created by this task.
+  volumes:
+  - name: workdir
+    emptyDir: { }
+  stepTemplate:
+    volumeMounts:
+    - mountPath: /var/workdir
+      name: workdir
+    workingDir: /var/workdir
+  steps:
+  - name: render-snapshot
+    image: registry.access.redhat.com/ubi9:latest@sha256:1057dab827c782abcfb9bda0c3900c0966b5066e671d54976a7bcb3a2d1a5e53
+    env:
+    - name: APPLICATION_NAME
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.labels['appstudio.openshift.io/application']
+    - name: COMPONENTS
+      value: $(params.COMPONENTS)
+    - name: NAMESPACE
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.namespace
+    - name: PIPELINE_RUN_NAME
+      valueFrom:
+        fieldRef:
+          fieldPath: metadata.labels['tekton.dev/pipelineRun']
+    - name: PRODUCT_VERSION
+      value: $(params.PRODUCT_VERSION)
+    - name: SNAPSHOT_NAME_RESULT_PATH
+      value: $(results.SNAPSHOT_NAME.path)
+    script: |
+      #!/usr/bin/env python3
+
+      import datetime
+      import json
+      import os
+      import re
+
+
+      def parse_components_input(raw_input):
+          return json.loads(raw_input)
+
+
+      def determine_resource_version_suffix(application_name):
+          match = re.search(r"(?P<version>-\d+-\d+)$", application_name)
+          if match:
+              return match.group("version")
+          return ""
+
+
+      def determine_snapshot_name(prefix, product_version):
+          # The timestamp is added to the Snapshot name so that we can differentiate Snapshots from rebuilds of the same commit or tag.
+          timestamp = datetime.datetime.now(datetime.timezone.utc).strftime("%Y%m%dT%H%M%SZ")
+          product_version_sanitized = product_version.replace(".", "-")
+          return f"{prefix}-{product_version_sanitized}-{timestamp}".lower()
```

Comment:
Oh, and one more thing! https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#dns-label-names says the name length is limited by 63. I'd suggest you take a substring of up to 63 characters of the composed name and return that.

Reply:
See https://github.com/stackrox/konflux-tasks/pull/22#discussion_r1905101113